### PR TITLE
Task/create tud interface

### PR DESCRIPTION
**Pull request by anzioka**
PR:

milestone 0 of the time use diary: 

1) interface to select activities during a full day duration and specify time spans for each activity
2) basic summary page showing each activity and time span, and ability to switch to a specific page in the PageMachine context and make changes.

Screen Shot 2020-07-29 at 9 58 48 PM
Screen Shot 2020-07-29 at 9 59 19 PM
Screen Shot 2020-07-29 at 9 58 30 PM

### CODE REVIEW

File: src/containers/tud/components/QuestionnaireForm.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  display: flex;
+  justify-content: space-between;
+  margin-top: 30px;
+`;
+
+const QuestionnaireForm = () => (
+  <Card>
+    <CardSegment>
+      <Paged
+          render={(pagedProps) => {
```
Comment (UnsungHero97) [inline src/containers/tud/components/QuestionnaireForm.js]: Is there a reason why the body for the `render` prop is not its own component?

File: src/containers/tud/components/QuestionnaireForm.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            if (page === 0) {
+              schema = PreSurveySchema.schema;
+              uiSchema = PreSurveySchema.uiSchema;
+            }
+
+            // day span
+            if (page === 1) {
+              schema = DaySpanSchema.schema;
+              uiSchema = DaySpanSchema.uiSchema;
+            }
```
Comment (UnsungHero97) [inline src/containers/tud/components/QuestionnaireForm.js]: A few thoughts...

First, if you use `if-else` blocks, you can avoid overwriting the schema variables on every render:

```
let dataSchema;
let uiSchema;

if (page === 0) {
  dataSchema = PreSurveySchema.schema;
  uiSchema = PreSurveySchema.uiSchema;
}
else if (page === 1) {
  dataSchema = DaySpanSchema.schema;
  uiSchema = DaySpanSchema.uiSchema;
}
else {
  dataSchema = createFormSchema(page, pagedData);
  uiSchema = createUiSchema(page);
}
```

Second, these assignments are dangerous. I believe any mutations on the left will also apply to the right. The way I've approached this in the past is to "clone" the schemas like so:

```
const dataSchema = JSON.parse(JSON.stringify(Schemas.dataSchema));
const uiSchema = JSON.parse(JSON.stringify(Schemas.uiSchema));
```

There are probably better ways to do the "clone"... let me know if you think of other ways.

File: src/containers/tud/components/QuestionnaireForm.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                      onClick={onBack}>
+                    Back
+                  </Button>
+                  <Button
+                      color="primary"
+                      onClick={handleNext}>
+                    {
+                      lastPage ? 'Submit' : 'Next'
+                    }
+                  </Button>
```
Comment (UnsungHero97) [inline src/containers/tud/components/QuestionnaireForm.js]: I know we're not submitting data right now, but in the future, these buttons must be disabled during the submit request, i.e. `SUBMIT_RS === RequestStates.PENDING`.

File: src/containers/tud/components/QuestionnaireForm.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+            const handleNext = () => {
+              //
+              validateAndSubmit();
+            };
+
+            const onChange = ({ formData } :Object) => {
+
+              const currentActivity = selectPrimaryActivityByPage(page, formData);
+              if (currentActivity) {
+                const endTimeInput = document.getElementById(`root_${getPageSectionKey(page, 0)}_endTime`);
```
Comment (UnsungHero97) [inline src/containers/tud/components/QuestionnaireForm.js]: Whoa... why do you need to do this?

File: src/containers/tud/components/TimeUseSummary.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  font-size: 15px;
+  grid-gap: 20px;
+  grid-template-columns: 100px 1fr;
+`;
+
+const Description = styled.p`
+  color: ${NEUTRAL.N600};
+  margin: 0;
+  padding: 0;
+`;
```
Comment (UnsungHero97) [inline src/containers/tud/components/TimeUseSummary.js:33]: How many components do you have in this project that are similar to this?

File: src/containers/tud/components/TimeUseSummary.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+import { createTimeUseSummary } from '../utils';
+
+const { NEUTRAL } = Colors;
+
+const Heading = styled.h6`
+  color: ${NEUTRAL.N900};
+  font-size: 16px;
+  font-weight: normal;
+  margin-bottom: 20px;
+`;
```
Comment (UnsungHero97) [inline src/containers/tud/components/TimeUseSummary.js:19]: How many components do you have in this project that are similar to this?

File: src/containers/tud/components/TimeUseSummary.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  display: grid;
+  font-size: 15px;
+  grid-gap: 20px;
+  grid-template-columns: 100px 1fr;
+`;
+
+const Description = styled.p`
+  color: ${NEUTRAL.N600};
+  margin: 0;
+  padding: 0;
```
Comment (UnsungHero97) [inline src/containers/tud/components/TimeUseSummary.js:32]: If you're going to nuke the margin and padding, might as well just use a `div` and avoid the extra css.

File: src/containers/tud/components/TimeUseSummary.js [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+                {item.time}
+              </Description>
+
+              <Description>
+                {item.description}
+              </Description>
+            </ItemSummary>
+            <StyledButton
+                onClick={() => goToPage(item.pageNum)}
+                startIcon={<FontAwesomeIcon icon={faPencilAlt} />}>
```
Comment (UnsungHero97) [inline src/containers/tud/components/TimeUseSummary.js]: I recommend moving the icon into a constant outside of the component.

```
const EditIcon = (
  <FontAwesomeIcon icon={faPencilAlt} />
);
```

File: src/containers/tud/constants/ActivitiesConstants.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
+// @flow
+
+// primary activities
+const CHILDCARE = 'Childcare/School';
+const EATING_DRINKING = 'Eating/Drinking';
+const GROOMING = 'Bathroom/Grooming';
+const INDOORS = 'In House (Chores, etc)';
+const MEDIA = 'Media Use (Tv, Books, Apps, etc)';
```
Comment (UnsungHero97) [inline src/containers/tud/constants/ActivitiesConstants.js]: `Tv` should be `TV`?

File: src/containers/tud/constants/ActivitiesConstants.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+  { name: SLEEPING, description: 'sleeping/resting', followup: true },
+  { name: EATING_DRINKING, description: 'eating/drinking', followup: true },
+  { name: GROOMING, description: 'grooming', followup: false },
+  { name: MEDIA, description: 'using media', followup: true },
+  { name: RECREATION_INSIDE, description: 'indoor recreation', followup: true },
+  { name: RECREATION_OUTSIDE, description: 'outdoor recreation', followup: true },
+  { name: CHILDCARE, description: 'child care', followup: false },
+  { name: OUTDOORS, description: 'travelling/running errands', followup: true },
+  { name: INDOORS, description: 'chores', followup: false },
+  { name: OTHER, description: '', followup: false }
```
Comment (UnsungHero97) [inline src/containers/tud/constants/ActivitiesConstants.js:38]: What is the `description` used for?

File: src/containers/tud/schemas/PreSurveySchema.js [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+          title: `We would like you to think about your child's day and complete the time use diary for yesterday.
+            What day of the week was yesterday?`,
+          // $FlowFixMe
+          enum: Info.weekdays(),
+          type: 'string'
+        },
+        [TYPICAL_DAY_FLAG]: {
+          title: `An important part of this project is to find out how children spend their time
+          during the week. Was yesterday a typical weekday for you and your child?
+          A non-typical day would include a school closing, being on vacation, or being home sick.`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/PreSurveySchema.js]: For multi-line strings like this, I recommend defining these as constants like so:

```
const TYPICAL_DAY_TITLE = 'An important part of this project is to find out how children spend their time'
  + ' during the week. Was yesterday a typical weekday for you and your child?'
  + ' A non-typical day would include a school closing, being on vacation, or being home sick.';
```

This way, you don't have to worry about new-line characters from the templated string and how the browser is going to render them. Additionally, it avoids submitting new-line characters, though I'm not entirely sure if that's the case to begin with.

If you really need the specific new-line cutoffs, I still recommend using a constant.

File: src/containers/tud/schemas/followup/EatingIndoorRecSchema.js [hunk 12]
```diff
@@ -0,0 +1,229 @@
...
+  MEDIA,
+} = PROPERTY_CONSTS;
+
+const { getPageSectionKey } = DataProcessingUtils;
+
+const bgAudioSchema = {
+  properties: {
+    [BG_AUDIO]: {
+      title: `Was there audio entertainment (e.g., music, talk radio)
+      on in the background while your child was doing this activity?`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/EatingIndoorRecSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/EatingIndoorRecSchema.js [hunk 13]
```diff
@@ -0,0 +1,229 @@
...
+      oneOf: [
+        {
+          properties: {
+            [BG_AUDIO]: {
+              enum: ['No', "Don't Know"]
+            },
+            [ADULT_MEDIA]: {
+              type: 'string',
+              title: `Was an adult using a tablet, laptop, cell phone/smart
+                phone at any point while your child did this activity?`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/EatingIndoorRecSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/EatingIndoorRecSchema.js [hunk 14]
```diff
@@ -0,0 +1,229 @@
...
+              items: {
+                type: 'string',
+                enum: ['Music', 'Talk Radio', 'Podcast', "Don't Know"]
+              },
+              uniqueItems: true
+            },
+            [ADULT_MEDIA]: {
+              type: 'string',
+              title: `Was an adult using a tablet, laptop, cell phone/smart
+                phone at any point while your child did this activity?`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/EatingIndoorRecSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/EatingIndoorRecSchema.js [hunk 15]
```diff
@@ -0,0 +1,229 @@
...
+      [LOCATION]: {
+        type: 'string',
+        title: 'Where was your child when he/she was doing this activity?',
+        enum: LOCATION_CATEGORIES
+      },
+      [MEDIA]: {
+        type: 'string',
+        title: `Was media being used at the same time as this activity, such as
+            television, movies, video or computer games,
+            books, magazines, radio or CDs, cell phone/smart phone, laptop or a tablet?`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/EatingIndoorRecSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/EatingIndoorRecSchema.js [hunk 16]
```diff
@@ -0,0 +1,229 @@
...
+        oneOf: [
+          {
+            properties: {
+              [MEDIA]: {
+                enum: ['No', "Don't Know"]
+              },
+              [ADULT_MEDIA]: {
+                type: 'string',
+                title: `Was an adult using a tablet, laptop, cell phone/smart
+                  phone at any point while your child did this activity?`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/EatingIndoorRecSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/EatingIndoorRecSchema.js [hunk 17]
```diff
@@ -0,0 +1,229 @@
...
+                  },
+                  {
+                    properties: {
+                      [BG_TV]: {
+                        enum: ['Yes']
+                      },
+                      // TODO: add otherText option to radio widget
+                      [BG_TV_AGE]: {
+                        title: `Was the program for your child's age, for older children,
+                        for younger children, or for adults?`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/EatingIndoorRecSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/MediaUseSchema.js [hunk 18]
```diff
@@ -0,0 +1,258 @@
...
+  PROGRAM_AGE,
+} = PROPERTY_CONSTS;
+
+const otherMediaSchema = {
+  properties: {
+    [OTHER_MEDIA]: {
+      type: 'string',
+      title: `Was any other media being used at the same time as this activity,
+      such as television, movies, video or computer games, books, magazines,
+      radio or CDs, cell phone/smart phone, laptop or a tablet? `,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/MediaUseSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/MediaUseSchema.js [hunk 19]
```diff
@@ -0,0 +1,258 @@
...
+    },
+    [LOCATION]: {
+      type: 'string',
+      title: 'Where was your child when he/she was doing this activity?',
+      enum: LOCATION_CATEGORIES
+    },
+    [ADULT_MEDIA]: {
+      type: 'string',
+      title: `Was an adult using a tablet, laptop, cell phone/smart phone
+        at any point while your child did this activity?`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/MediaUseSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/MediaUseSchema.js [hunk 20]
```diff
@@ -0,0 +1,258 @@
...
+                  'Social call', 'Entertainment', 'Social media']
+              },
+              uniqueItems: true,
+              minItems: 1
+            },
+            [ADULT_MEDIA_PROPORTION]: {
+              type: 'number',
+              title: `Approximately what percentage of the time for this activity were you using your device?
+              For example, if your child was using media for 2 hours and you used your device for 1 hour,
+               enter 50; if you used your device the entire time, enter 100.`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/MediaUseSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/MediaUseSchema.js [hunk 21]
```diff
@@ -0,0 +1,258 @@
...
+        oneOf: [
+          {
+            properties: {
+              [MEDIA_ACTIVITY]: {
+                enum: MEDIA_ACTIVITY_CATEGORIES.slice(0, 4)
+              },
+              [PROGRAM_AGE]: {
+                type: 'string',
+                title: `Was the program (e.g., show, app, movie, book, etc.) for your child's age,
+                  for older children, for younger children, or for adults?`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/MediaUseSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/MediaUseSchema.js [hunk 22]
```diff
@@ -0,0 +1,258 @@
...
+                enum: ["Child's age", 'Older children', 'Younger children', 'Adults', "Don't know/other"]
+              },
+              [MEDIA_NAME]: {
+                type: 'string',
+                title: `What was the name of the media used? (e.g. name of show, book, movie).
+                 If you are unsure of exact name, either describe (e.g., a drawing app, a movie about unicorns)
+                 or leave blank`
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/MediaUseSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/OutdoorRecSchema.js [hunk 23]
```diff
@@ -0,0 +1,112 @@
...
+          enum: CAREGIVERS
+        },
+        uniqueItems: true,
+        minItems: 1
+      },
+      [MEDIA]: {
+        type: 'string',
+        title: `Was media being used at the same time as this activity, such as
+            television, movies, video or computer games,
+            books, magazines, radio or CDs, cell phone/smart phone, laptop or a tablet?`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/OutdoorRecSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/OutdoorRecSchema.js [hunk 24]
```diff
@@ -0,0 +1,112 @@
...
+      [MEDIA]: {
+        oneOf: [
+          {
+            properties: {
+              [MEDIA]: {
+                enum: ['Yes']
+              },
+              [BG_AUDIO]: {
+                title: `Was there audio entertainment (e.g., music, talk radio)
+                on in the background while your child was doing this activity?`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/OutdoorRecSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/OutdoorRecSchema.js [hunk 25]
```diff
@@ -0,0 +1,112 @@
...
+                type: 'string',
+                enum: ['Yes', 'No', "Don't Know"]
+              },
+              [ADULT_MEDIA]: {
+                type: 'string',
+                title: `Was an adult using a tablet, laptop, cell phone/smart
+                  phone at any point while your child did this activity?`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/OutdoorRecSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/OutdoorRecSchema.js [hunk 26]
```diff
@@ -0,0 +1,112 @@
...
+          },
+          {
+            properties: {
+              [MEDIA]: {
+                enum: ['No', "Don't Know"]
+              },
+              [ADULT_MEDIA]: {
+                type: 'string',
+                title: `Was an adult using a tablet, laptop, cell phone/smart
+                  phone at any point while your child did this activity?`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/OutdoorRecSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/OutdoorsSchema.js [hunk 27]
```diff
@@ -0,0 +1,186 @@
...
+} = PROPERTY_CONSTS;
+
+const { getPageSectionKey } = DataProcessingUtils;
+
+const adultMediaSchema = {
+  properties: {
+    [ADULT_MEDIA]: {
+      type: 'string',
+      title: `Was an adult using a tablet, laptop, cell phone/smart phone
+        at any point while your child did this activity?`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/OutdoorsSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/OutdoorsSchema.js [hunk 28]
```diff
@@ -0,0 +1,186 @@
...
+            [ADULT_MEDIA]: {
+              enum: ['Yes']
+            },
+            [ADULT_MEDIA_PURPOSE]: {
+              type: 'array',
+              title: 'What was the tablet/laptop/cell phone/smart phone used for? Please choose all that apply',
+              items: {
+                type: 'string',
+                enum: ['Work call', 'Work email', 'Review of documents for work',
+                  'Social call', 'Entertainment', 'Social media']
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/OutdoorsSchema.js:43]: Formatting. Either all on one line, or each on its own line.

File: src/containers/tud/schemas/followup/OutdoorsSchema.js [hunk 29]
```diff
@@ -0,0 +1,186 @@
...
+      [SECONDARY_LOCATION]: {
+        type: 'string',
+        title: 'Where was your child when he/she was doing this activity?',
+        enum: SECONDARY_LOCATIONS
+      },
+      [MEDIA]: {
+        type: 'string',
+        title: `Was media being used at the same time as this activity, such as
+            television, movies, video or computer games,
+            books, magazines, radio or CDs, cell phone/smart phone, laptop or a tablet?`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/OutdoorsSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/OutdoorsSchema.js [hunk 30]
```diff
@@ -0,0 +1,186 @@
...
+        oneOf: [
+          {
+            properties: {
+              [MEDIA]: {
+                enum: ['Yes']
+              },
+              [MEDIA_AGE]: { // TODO: otherText radio
+                type: 'array',
+                title: `Was the media content directed at your child's age,
+                  older children, younger children, or adults?`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/OutdoorsSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/OutdoorsSchema.js [hunk 31]
```diff
@@ -0,0 +1,186 @@
...
+                items: {
+                  type: 'string',
+                  enum: ["Child's age", 'Older children', 'Younger children', 'Adults']
+                },
+                uniqueItems: true
+              },
+              [BG_AUDIO]: {
+                title: `Was there audio entertainment (e.g., music, talk radio)
+                on in the background while your child was doing this activity?`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/OutdoorsSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/SleepingSchema.js [hunk 32]
```diff
@@ -0,0 +1,214 @@
...
+        type: 'string',
+        enum: ['Music', 'Talk Radio', 'Podcast', "Don't Know"]
+      },
+      uniqueItems: true
+    },
+    [BG_MEDIA_PROPORTION]: {
+      title: `Approximately what percentage of time that the child was
+      sleeping was the background media in use?
+      For example, if your child slept for 10 hours and the radio was on for 1,
+       enter 10; if on the full time, enter 100.`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/SleepingSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/SleepingSchema.js [hunk 33]
```diff
@@ -0,0 +1,214 @@
...
+
+const createSchema = (pageNum :number) => ({
+  [getPageSectionKey(pageNum, 1)]: {
+    type: 'object',
+    title: '',
+    properties: {
+      [MEDIA]: {
+        title: `Was media being used in the same room as the child while they were sleeping?
+          This could include television, movies, video or computer games, radio or CDs,
+          but does not include white noise machines. `,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/SleepingSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/SleepingSchema.js [hunk 34]
```diff
@@ -0,0 +1,214 @@
...
+              [BG_TV]: {
+                oneOf: [
+                  {
+                    properties: {
+                      [BG_TV]: {
+                        enum: ['No', "Don't Know"]
+                      },
+                      [BG_AUDIO]: {
+                        title: `Was there audio entertainment (e.g., music, talk radio)
+                        on in the background while your child was doing this activity?`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/SleepingSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/SleepingSchema.js [hunk 35]
```diff
@@ -0,0 +1,214 @@
...
+                    }
+                  },
+                  {
+                    properties: {
+                      [BG_TV]: {
+                        enum: ['Yes']
+                      },
+                      [BG_TV_AGE]: {
+                        title: `Was the program for your child's age, for older children,
+                        for younger children, or for adults?`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/SleepingSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/SleepingSchema.js [hunk 36]
```diff
@@ -0,0 +1,214 @@
...
+                        type: 'array',
+                        items: {
+                          type: 'string',
+                          enum: ["Child's age", 'Older children', 'Younger children', 'Adults'],
+                        },
+                        uniqueItems: true
+                      },
+                      [BG_AUDIO]: {
+                        title: `Was there audio entertainment (e.g., music, talk radio)
+                        on in the background while your child was doing this activity?`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/SleepingSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/schemas/followup/SleepingSchema.js [hunk 37]
```diff
@@ -0,0 +1,214 @@
...
+                          {
+                            properties: {
+                              [BG_AUDIO]: {
+                                enum: ['No', "Don't Know"]
+                              },
+                              [BG_MEDIA_PROPORTION]: {
+                                title: `Approximately what percentage of time that the child was
+                                sleeping was the background media in use?
+                                For example, if your child slept for 10 hours and the radio was on for 1,
+                                 enter 10; if on the full time, enter 100.`,
```
Comment (UnsungHero97) [inline src/containers/tud/schemas/followup/SleepingSchema.js]: Same comment about multi-line strings.

File: src/containers/tud/utils/index.js [hunk 38]
```diff
@@ -0,0 +1,261 @@
...
+  const psk = getPageSectionKey(pageNum, 0);
+
+  const activityName = getIn(formData, [psk, ACTIVITY_NAME]);
+  return PRIMARY_ACTIVITIES.find((activity) => activity.name === activityName);
+};
+
+const createFormSchema = (pageNum :number, formData :Object) => {
+
+  const prevActivity = selectPrimaryActivityByPage(pageNum - 1, formData);
+  const prevEndTime = pageNum === 2
```
Comment (UnsungHero97) [inline src/containers/tud/utils/index.js]: `2` is an important value and should be defined as a constant somewhere.

File: src/containers/tud/utils/index.js [hunk 39]
```diff
@@ -0,0 +1,261 @@
...
+    title: '',
+    properties: {
+      [getPageSectionKey(pageNum, 0)]: {
+        properties: {
+          [ACTIVITY_NAME]: {
+            type: 'string',
+            title: (pageNum === 2
+              ? `What did your child start doing at ${formattedTime}?`
+              : `What time did your child start doing at ${formattedTime} after they
+                finished ${(prevActivity || {}).description}?`),
```
Comment (UnsungHero97) [inline src/containers/tud/utils/index.js]: Same comment about multi-line strings.

File: src/containers/tud/utils/index.js [hunk 40]
```diff
@@ -0,0 +1,261 @@
...
+          [ACTIVITY_START_TIME]: {
+            type: 'string',
+            title: '',
+            default: prevEndTime.toLocaleString(DateTime.TIME_24_SIMPLE)
+          },
+          [ACTIVITY_END_TIME]: {
+            id: 'end_time',
+            type: 'string',
+            title: currentActivity ? `When did your child stop ${currentActivity.description}?`
+              : 'When did the selected activity end?'
```
Comment (UnsungHero97) [inline src/containers/tud/utils/index.js]: Formatting. `?` and `:` should be on their own lines.

```
title: currentActivity
  ? `When...`
  : 'When...'
```

File: src/containers/tud/utils/index.js [hunk 41]
```diff
@@ -0,0 +1,261 @@
...
+    MediaUseSchema.createUiSchema(pageNum),
+    OutdoorRecSchema.createUiSchema(pageNum),
+    OutdoorsSchema.createUiSchema(pageNum)
+  );
+
+  return {
+    [getPageSectionKey(pageNum, 0)]: {
+      classNames: 'column-span-12 grid-container',
+      [ACTIVITY_NAME]: {
+        classNames: (pageNum === 1 ? 'hidden' : 'column-span-12')
```
Comment (UnsungHero97) [inline src/containers/tud/utils/index.js]: `1` is an important value and should be defined as a constant somewhere.

**Review by UnsungHero97 — CHANGES_REQUESTED** [review]

File: src/containers/tud/components/QuestionnaireForm.js (hunk 2 above)
Comment (anzioka) [reply src/containers/tud/components/QuestionnaireForm.js]: Can you expound on your second point? i.e what do you mean by mutations on the left will also apply to the right?

File: src/containers/tud/components/QuestionnaireForm.js (hunk 3 above)
Comment (anzioka) [reply src/containers/tud/components/QuestionnaireForm.js]: Noted.

File: src/containers/tud/components/QuestionnaireForm.js (hunk 4 above)
Comment (anzioka) [reply src/containers/tud/components/QuestionnaireForm.js]: After selecting a primary activity, the label on the next question needs to be updated to reference the selected activity. For example, after selecting `Sleeping/Resting` the next question would be `When did your child stop sleeping/resting?`

File: src/containers/tud/constants/ActivitiesConstants.js (hunk 10 above)
Comment (anzioka) [reply src/containers/tud/constants/ActivitiesConstants.js:38]: Usage:

`When did your child stop ${activity.description}?`

File: src/containers/tud/utils/index.js [hunk 42]
```diff
@@ -0,0 +1,265 @@
...
+    if (currentEndTime <= currentStartTime) {
+      errors[psk][endTimeKey].addError(errorMsg);
+    }
+    // the last activity of the day should end at the time the child went to bed
+    // $FlowFixMe invalid-compare
+    if (currentEndTime > dayEndTime) {
+      errors[psk][endTimeKey].addError(`The last activity of the
+          day should end at ${dayEndTime.toLocaleString(DateTime.TIME_SIMPLE)}
+          since you indicated the child went to bed then.`);
+    }
```
Comment (solomontang) [inline src/containers/tud/utils/index.js:252]: There is a way to have the date/time picker also set min/max ranges. we should investigate this with fabricate moving forward.

File: src/containers/tud/schemas/followup/OutdoorsSchema.js [hunk 43]
```diff
@@ -0,0 +1,182 @@
...
+    required: [CAREGIVER, SECONDARY_LOCATION, MEDIA],
+    dependencies: {
+      [MEDIA]: {
+        oneOf: [
+          {
+            properties: {
+              [MEDIA]: {
+                enum: ['Yes']
+              },
+              [MEDIA_AGE]: { // TODO: otherText radio
```
Comment (solomontang) [inline src/containers/tud/schemas/followup/OutdoorsSchema.js]: Is this TODO still relevant

**Review by solomontang — DISMISSED** [review]

**Comment by solomontang** [comment]
damn that's soo much padding..

File: src/containers/tud/components/QuestionnaireForm.js (hunk 2 above)
Comment (solomontang) [reply src/containers/tud/components/QuestionnaireForm.js]: He's suggesting a defensive copy of the schema if for some reason any mutations to dataSchema causes PreSurveySchema to change. Could always make schemas immutable.

Comment (anzioka) [reply src/containers/tud/components/QuestionnaireForm.js]: Ok, makes sense! 

in that case, I think I could use lodash cloneDeep (https://lodash.com/docs/4.17.15#cloneDeep) to make a deep clone

File: src/containers/tud/components/QuestionnaireForm.js [hunk 44]
```diff
@@ -0,0 +1,144 @@
...
+  let uiSchema;
+
+  if (page === PRE_SURVEY_PAGE) {
+    schema = cloneDeep(PreSurveySchema.schema);
+    uiSchema = cloneDeep(PreSurveySchema.uiSchema);
+  }
+  else if (page === DAY_SPAN_PAGE) {
+    schema = cloneDeep(DaySpanSchema.schema);
+    uiSchema = cloneDeep(DaySpanSchema.uiSchema);
+  }
```
Comment (solomontang) [inline src/containers/tud/components/QuestionnaireForm.js:62]: I see you've gone the route of making a deep copy. I wonder if you could have gotten away with Object.freeze when declaring the schemas.

File: src/containers/tud/components/QuestionnaireForm.js [hunk 45]
```diff
@@ -0,0 +1,144 @@
...
+  else {
+    schema = createFormSchema(page, pagedData);
+    uiSchema = createUiSchema(page);
+  }
+
+  const handleNext = () => {
+    //
```
Comment (solomontang) [inline src/containers/tud/components/QuestionnaireForm.js:69]: rogue comment

**Review by solomontang — APPROVED** [review]

File: src/containers/tud/components/QuestionnaireForm.js (hunk 44 above)
Comment (anzioka) [reply src/containers/tud/components/QuestionnaireForm.js:62]: ah yes, I think that would have worked too.

**Review by UnsungHero97 — APPROVED** [review]

**Pull request merged**